bazel/external/boringssl_fips.genrule_cmd

```diff
@@ -0,0 +1,95 @@
+#!/bin/bash
+
+set -e
+
+# BoringSSL build as described in the Security Policy for BoringCrypto module (2018-10-25):
+# https://csrc.nist.gov/CSRC/media/projects/cryptographic-module-validation-program/documents/security-policies/140sp3318.pdf
+
+# This works only on Linux-x86_64.
+if [[ `uname` != "Linux" || `uname -m` != "x86_64" ]]; then
+  echo "ERROR: BoringSSL FIPS is currently supported only on Linux-x86_64."
+  exit 1
+fi
+
+# Bazel magic.
+ROOT=$$(dirname $(rootpath boringssl/BUILDING.md))/..
+pushd $$ROOT
+
+# Build tools requirements:
+# - Clang compiler version 6.0.1 (http://releases.llvm.org/download.html)
+# - Go programming language version 1.10.3 (https://golang.org/dl/)
+# - Ninja build system version 1.8.2 (https://github.com/ninja-build/ninja/releases)
+
+# Override $$PATH for build tools, to avoid picking up anything else.
+export PATH="$$(dirname `which cmake`):/usr/bin:/bin"
+
+# Clang 6.0.1
```

Comment:
FWIW, I think it would be ideal if we could use a consistent toolchain across the entire build. We've had issues in the past with string libraries and compiler/glibc couplings that would be simplified here. Can't we just mandate that the user actually builds under clang-6.0.1 if they want a real FIPS build?

Reply:
AFAIK, the issue that you're referring to is/was with linking against versions of libstdc++ that provide different ABIs, not using different compilers.

There are no issues with linking libraries produced using different toolchains (in fact, we're doing this already, since all prebuilt libraries are always built using gcc... same goes for the system libraries), and enforcing older toolchain on rest of the build is going to result in arguably "worse" binary, for no good reason, IMHO.

Yes, we could mandate that users build using clang-6.0.1, go-1.10.3 and ninja-1.8.2, and fail the build otherwise, but then we're only making things harder for the users.

Reply:
I don't see a strong reason to mandate users build using clang-6.0.1, go-1.10.3 and ninja-1.8.2, we might be able to skip download if existing toolchains matches the version we expected. It will make also make CI harder if we want FIPS build with different build image.

Btw, is it possible to use Bazel managed Go binary in a genrule?